pkg/alertmanager/operator.go

```diff
@@ -821,6 +821,9 @@ func createSSetInputHash(a monitoringv1.Alertmanager, c Config, tlsAssets *opera
 		http2 = a.Spec.Web.WebConfigFileFields.HTTPConfig.HTTP2
 	}
 
+	// Ignoring changes to RevisionHistoryLimit field
```

Comment:
(suggestion) provide a bit more context.
```suggestion
	// The controller should ignore any changes to RevisionHistoryLimit field because 
	// it may be modified by external actors.
	// See https://github.com/prometheus-operator/prometheus-operator/issues/5712
```